### PR TITLE
Move IsBuildOnlyDiagnostic to use a switch expression

### PR DESCRIPTION
**Pull request by 333fred**
By suppressing CS8524, we can get a build-time diagnostic when we forget to update this, rather than getting a test failure that we often forget until CI throws.

### CODE REVIEW

**Review by 333fred — COMMENTED** [review]
GitHub's online view makes it extremely hard to verify that no changes were made here. Highly recommend using vscode or (I assume) codeflow to make the review easier. @dotnet/roslyn-compiler for review, this is the change we discussed in standup.

**Review by jcouv — APPROVED** [review]
Diff in CodeFlow was sane ;-) LGTM Thanks (iteration 1) modulo codegen confirmation

File: src/Compilers/CSharp/Portable/Errors/ErrorFacts.cs [hunk 1]
```diff
@@ -573,1877 +573,1875 @@ internal static int GetWarningLevel(ErrorCode code)
...
+                or ErrorCode.WRN_NullabilityMismatchInParameterTypeOnInterceptor
+                or ErrorCode.ERR_InterceptorCannotInterceptNameof
+                or ErrorCode.ERR_SymbolDefinedInAssembly
+                or ErrorCode.ERR_InterceptorArityNotCompatible
+                or ErrorCode.ERR_InterceptorCannotBeGeneric
+                or ErrorCode.ERR_InterceptableMethodMustBeOrdinary
+                or ErrorCode.ERR_PossibleAsyncIteratorWithoutYield
+                or ErrorCode.ERR_PossibleAsyncIteratorWithoutYieldOrAwait
+                    => true, // Update src\EditorFeatures\CSharp\LanguageServer\CSharpLspBuildOnlyDiagnostics.cs
+                             // whenever new values are added here.
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Errors/ErrorFacts.cs:627]: nit: I subjectively find the below suggestion more readable, use it or don't at your discrection.

```suggestion
                    // Update src\EditorFeatures\CSharp\LanguageServer\CSharpLspBuildOnlyDiagnostics.cs
                    // whenever new values are added here.
                    => true,
```

**Review by RikkiGibson — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Errors/ErrorFacts.cs [hunk 2]
```diff
@@ -573,1877 +573,1875 @@ internal static int GetWarningLevel(ErrorCode code)
...
+                or ErrorCode.ERR_ParamsCollectionExtensionAddMethod
+                or ErrorCode.ERR_ParamsCollectionMissingConstructor
+                or ErrorCode.ERR_NoModifiersOnUsing
+                or ErrorCode.ERR_CannotDynamicInvokeOnExpression
+                or ErrorCode.ERR_InterceptsLocationDataInvalidFormat
+                or ErrorCode.ERR_InterceptsLocationUnsupportedVersion
+                or ErrorCode.ERR_InterceptsLocationDuplicateFile
+                or ErrorCode.ERR_InterceptsLocationFileNotFound
+                or ErrorCode.ERR_InterceptsLocationDataInvalidPosition
+                or ErrorCode.INF_TooManyBoundLambdas
```
[the remark this answers is not in the document]
Comment (jjonescz) [reply src/Compilers/CSharp/Portable/Errors/ErrorFacts.cs:2442]: > Mac leg starts failing

mac leg is already failing for a long time :D - https://github.com/dotnet/roslyn/issues/70233

**Pull request merged**